halibot/halauth.py

```diff
@@ -0,0 +1,77 @@
+import json
+import logging
+
+def hasPermission(perm, reply=False):
+	def real_dec(func):
+		def wrapper(self, msg, *args, **kwargs):
+			if self._hal.auth.hasPermission(msg.origin, msg.author, perm):
+				func(self, msg, *args, **kwargs)
+			elif reply:
+				self.reply(msg, body="Permission Denied")
```

Comment:
Thoughts on `self.config.get("no-perm","Permission Denied")` ? Probably use a better config key than `no-perm`, but it would be an optional key on all modules.

Alternatively, having a halibot-global option.

Also, the `reply` bool should probably be a config option, not at the annotation level